packages/react/src/components/LocaleModal/LocaleModal.js

```diff
@@ -71,6 +71,21 @@ const LocaleModal = ({ isOpen, setIsOpen }) => {
     }
   }, [clearResults]);
 
+  useEffect(() => {
+    /**
+     * @param {number} keyCode - The code of the keyboard key pressed.
+     */
+    function closeFilterOnEscapeKeyPress({ keyCode }) {
+      if (keyCode === 27 && isFiltering) {
+        setIsFiltering(false);
+      }
+    }
```

Comment:
Minor thing, let's make this a private method outside of the `useEffect`, and include the underscore in front of the method name, maybe something like `_closeModalOnEscape`. Be sure to include the `@private` flag in the JSDocs.